src/metabase/util/log.clj

```diff
@@ -0,0 +1,187 @@
+         (let [x# ~x]
+           (if (instance? Throwable x#)
+             (clojure.tools.logging/log* logger# ~level x#  (format ~@more))
+             (clojure.tools.logging/log* logger# ~level nil (format x# ~@more))))))))
+
+;;; ------------------------------------------------ Internal macros -------------------------------------------------
+(defmacro logp
+  "Implementation for prn-style `logp`.
+  You shouldn't have to use this directly; prefer the level-specific macros like [[info]]."
+  {:arglists '([level message & more] [level throwable message & more])}
+  [level x & more]
+  (macros/case
+    :cljs (glogi-logp (str *ns*) level x more)
+    :clj  (tools-logp *ns*       level x more)))
+
+(defmacro logf
+  "Implementation for printf-style `logf`.
+  You shouldn't have to use this directly; prefer the level-specific macros like [[infof]]."
+  [level x & args]
+  (macros/case
+    :cljs (glogi-logf (str *ns*) level x args)
+    :clj  (tools-logf *ns*       level x args)))
+
+;;; --------------------------------------------------- Public API ---------------------------------------------------
+(defmacro trace
+  "Log one or more args at the `:trace` level."
+  {:arglists '([& args] [e & args])}
+  [& args]
+  `(logp :trace ~@args))
+
+(defmacro tracef
+  "Log a message at the `:trace` level by applying `format` to a format string and args."
+  {:arglists '([format-string & args] [e format-string & args])}
+  [& args]
+  `(logf :trace ~@args))
+
+(defmacro debug
+  "Log one or more args at the `:debug` level."
+  {:arglists '([& args] [e & args])}
+  [& args]
+  `(logp :debug ~@args))
+
+(defmacro debugf
+  "Log a message at the `:debug` level by applying `format` to a format string and args."
+  {:arglists '([format-string & args] [e format-string & args])}
+  [& args]
+  `(logf :debug ~@args))
+
+(defmacro info
+  "Log one or more args at the `:info` level."
+  {:arglists '([& args] [e & args])}
+  [& args]
+  `(logp :info ~@args))
+
+(defmacro infof
+  "Log a message at the `:info` level by applying `format` to a format string and args."
+  {:arglists '([format-string & args] [e format-string & args])}
+  [& args]
+  `(logf :info ~@args))
+
+(defmacro warn
+  "Log one or more args at the `:warn` level."
+  {:arglists '([& args] [e & args])}
+  [& args]
+  `(logp :warn ~@args))
+
+(defmacro warnf
+  "Log a message at the `:warn` level by applying `format` to a format string and args."
+  {:arglists '([format-string & args] [e format-string & args])}
+  [& args]
+  `(logf :warn ~@args))
+
+(defmacro error
+  "Log one or more args at the `:error` level."
+  {:arglists '([& args] [e & args])}
+  [& args]
+  `(logp :error ~@args))
+
+(defmacro errorf
+  "Log a message at the `:error` level by applying `format` to a format string and args."
+  {:arglists '([format-string & args] [e format-string & args])}
+  [& args]
+  `(logf :error ~@args))
+
+;; TODO Uncomment this when it's needed.
+#_(defmacro fatal
```

Comment:
Why is this commented out?

Reply:
There are no users of it anywhere. I'd rather not delete it, hence it's commented out and has a TODO. Should I rephrase the TODO to make it more clear?

Reply:
I'll uncomment this and ignore the Kondo warning.